app/scripts/modules/core/src/serverGroup/ServerGroup.tsx

```diff
@@ -0,0 +1,272 @@
+import * as React from 'react';
+import * as ReactGA from 'react-ga';
+import { has, get } from 'lodash';
+import classNames = require('classnames');
+import autoBindMethods from 'class-autobind-decorator';
+import { Subscription } from 'rxjs/Subscription';
+
+import { Application } from 'core/application';
+import { CloudProviderLogo } from 'core/cloudProvider';
+import { serverGroupSequenceFilter } from 'core/cluster/serverGroup.sequence.filter';
+import { IInstance, IServerGroup } from 'core/domain';
+import { EntityNotifications } from 'core/entityTag/notifications/EntityNotifications';
+import { HealthCounts } from 'core/healthCounts';
+import { LoadBalancersTag } from 'core/loadBalancer/LoadBalancersTag';
+import { NamingService } from 'core/naming';
+import { NgReact, ReactInjector } from 'core/reactShims'
+import { Instances } from 'core/instance/Instances';
+import { ScrollToService } from 'core/utils';
+
+export interface JenkinsViewModel {
+  number: number;
+  href?: string;
+}
+
+export interface IServerGroupProps {
+  cluster: string;
+  serverGroup: IServerGroup;
+  application: Application;
+  sortFilter: any;
+  hasLoadBalancers: boolean;
+  hasDiscovery: boolean;
+}
+
+export interface IServerGroupState {
+  serverGroup: IServerGroup;
+  serverGroupSequence: string;
+  jenkins: JenkinsViewModel;
+  hasBuildInfo: boolean;
+  instances: IInstance[];
+  images?: string;
+
+  filter: string;
+  showAllInstances: boolean;
+  listInstances: boolean;
+
+  multiselect: boolean;
+  isSelected: boolean; // single select mode
+  isMultiSelected: boolean; // multiselect mode
+}
+
+const getSequence = serverGroupSequenceFilter(new NamingService());
+
+@autoBindMethods
+export class ServerGroup extends React.Component<IServerGroupProps, IServerGroupState> {
+  private stateChangeSubscription: Subscription;
+  private serverGroupsSubscription: Subscription;
+
+  constructor(props: IServerGroupProps) {
+    super(props);
+    this.state = this.getState(props);
+  }
+
+  private getState(props: IServerGroupProps): IServerGroupState {
+    const { serverGroup } = props;
+    const { showAllInstances, listInstances, multiselect, filter } = props.sortFilter;
+    const instances = serverGroup.instances.filter(i => ReactInjector.clusterFilterService.shouldShowInstance(i));
+    const serverGroupSequence = getSequence(serverGroup.name);
+    const hasBuildInfo = !!serverGroup.buildInfo;
+    const isSelected = this.isSelected(serverGroup);
+    const isMultiSelected = this.isMultiSelected(multiselect, serverGroup);
+    const jenkinsConfig = serverGroup.buildInfo && serverGroup.buildInfo.jenkins;
+
+    let jenkins: JenkinsViewModel = null;
```

Comment:
Definitely not the PR for this, but we should probably pull out anything Jenkins specific and wrap it in some kind of `BuildViewModel`, especially if we're trying to move people off of Jenkins in the somewhat foreseeable future.